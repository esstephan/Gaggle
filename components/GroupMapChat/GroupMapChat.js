import React, { Component } from 'react';
import { Container, Content, Tab, Tabs, TabHeading, Icon, Text, Header } from 'native-base';
import Chat from '../Chat/Chat.js';
import MapView from '../MapDisplay/MapDisplay.js';

export default class GroupMapChat extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: this.props.user
    }
  }

  render() {
    return (
      <Container>
        <Header hasTabs />
        <Content>
          <Tabs>
            <Tab heading={ <TabHeading><Icon name="compass" /><Text>Map</Text></TabHeading>}>
              <View>
                <MapDisplay />
              </View>
            </Tab>
            <Tab heading={ <TabHeading><Icon name="chatboxes" /><Text>Chat</Text></TabHeading>}>
              <Chat user={this.state.user} />
            </Tab>
          </Tabs>
        </Content>
      </Container>
    )
  }
}
