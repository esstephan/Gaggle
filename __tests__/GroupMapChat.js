import 'react-native';
import React from 'react';
import GroupMapChat from '../components/GroupMapChat/GroupMapChat.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

xit('renders correctly', () => {
  const tree = renderer.create(
    <GroupMapChat />
  );
  expect(tree).toMatchSnapshot();
});