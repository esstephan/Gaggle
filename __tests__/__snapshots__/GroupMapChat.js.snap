exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}>
  <View>
    <View
      hasTabs={true}
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": null,
          "elevation": 0,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 15,
          "right": 0,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": 0,
        }
      }
      virtual={undefined} />
  </View>
  <RCTScrollView>
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }>
        <View
          style={
            Object {
              "borderBottomWidth": 0.5,
              "borderColor": "#a7a6ab",
              "elevation": 3,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-around",
            }
          }
          virtual={undefined}>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}>
            <View
              active={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F8F8F8",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
              virtual={undefined}>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 26,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}>
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#007aff",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "marginLeft": 7,
                    "marginRight": 7,
                  }
                }
                virtual={undefined}>
                Map
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}>
            <View
              active={false}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F8F8F8",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
              virtual={undefined}>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#6b6b6b",
                      "fontSize": 26,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
                virtual={undefined}>
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#6b6b6b",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "marginLeft": 7,
                    "marginRight": 7,
                  }
                }
                virtual={undefined}>
                Chat
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#007aff",
                "bottom": 0,
                "height": 4,
                "left": 0,
                "position": "absolute",
                "width": 375,
              }
            } />
        </View>
        <RCTScrollView>
          <View>
            <View
              style={
                Object {
                  "width": 750,
                }
              }>
              <View
                heading={
                  <Styled(TabHeadingNB)
                    virtual={undefined}>
                    <Styled(Icon)
                      name="compass"
                      virtual={undefined} />
                    <Styled(Text)
                      virtual={undefined}>
                      Map
                    </Styled(Text)>
                  </Styled(TabHeadingNB)>
                }
                style={
                  Object {
                    "backgroundColor": "#FFF",
                    "flex": 1,
                  }
                }
                virtual={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    }
                  }
                  virtual={undefined}>
                  Map goes here
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "width": 750,
                }
              }>
              <View
                heading={
                  <Styled(TabHeadingNB)
                    virtual={undefined}>
                    <Styled(Icon)
                      name="chatboxes"
                      virtual={undefined} />
                    <Styled(Text)
                      virtual={undefined}>
                      Chat
                    </Styled(Text)>
                  </Styled(TabHeadingNB)>
                } />
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
